### PR TITLE
Feature/flyweight

### PR DESCRIPTION
**Pull request by lukasziukas**
Flyweight, minor fixes, disposed objects pool, power ups spawn(need fixes, big commit)

### CODE REVIEW

File: Core/NoOpRunner.Core/Enums/Entity.cs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿namespace NoOpRunner.Core.Enums
 {
-    public enum Entity
+    public enum Entity//meh?
```
Comment (Kodnot) [inline Core/NoOpRunner.Core/Enums/Entity.cs]: Elaborate? xD

Comment (lukasziukas) [reply Core/NoOpRunner.Core/Enums/Entity.cs]: MUDA

File: Core/NoOpRunner.Core/PowerUpsContainer.cs [hunk 2]
```diff
@@ -13,23 +13,36 @@ namespace NoOpRunner.Core
     /// </summary>
     public class PowerUpsContainer : ShapesContainer, IObserver
     {
+        private readonly Random random = new Random(DateTime.Now.Millisecond);
```
Comment (Kodnot) [inline Core/NoOpRunner.Core/PowerUpsContainer.cs]: We have the RandomNumber singleton, could use that

File: Core/NoOpRunner.Core/Enums/Entity.cs (hunk 1 above)
Comment (lukasziukas) [reply Core/NoOpRunner.Core/Enums/Entity.cs]: Useless now, maybe use for UI type?

File: Core/NoOpRunner.Core/Shapes/BaseShape.cs [hunk 3]
```diff
@@ -91,12 +93,12 @@ public virtual WindowPixelCollection Render()
             return windowPixelCollection;
         }
 
-        public bool IsHit(int x, int y)
+        public bool IsHit(int x, int y)//meh?
```
Comment (Kodnot) [inline Core/NoOpRunner.Core/Shapes/BaseShape.cs:96]: Will still mayyyyyyyyyybe be used when proper collision handling is implemented

**Review by Kodnot — APPROVED** [review]

File: Core/NoOpRunner.Core/Enums/Entity.cs (hunk 1 above)
Comment (Kodnot) [reply Core/NoOpRunner.Core/Enums/Entity.cs]: You could probably safely remove anything unused, it's just an enum, if we need it we can reintroduce it

File: Core/NoOpRunner.Core/PowerUpsContainer.cs (hunk 2 above)
Comment (lukasziukas) [reply Core/NoOpRunner.Core/PowerUpsContainer.cs]: Yes

File: Core/NoOpRunner.Core/Enums/Entity.cs (hunk 1 above)
Comment (lukasziukas) [reply Core/NoOpRunner.Core/Enums/Entity.cs]: Removed

**Pull request merged**